Network timeout settings for wallabug plugin

zikasak opened the pull request:
Added an ability to set network timeouts manually (for example, in case of slow wallabag server). 

![FileManager_2025-05-14_200617](https://github.com/user-attachments/assets/b8b51c1f-2ff8-4423-8f7c-259d60ebd57e)
![FileManager_2025-05-14_200627](https://github.com/user-attachments/assets/f4ac6a1f-1101-415a-aa7f-e45c18478adb)



Closes https://github.com/koreader/koreader/issues/9101

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/koreader/koreader/13786)
<!-- Reviewable:end -->

Frenzie commented on plugins/wallabag.koplugin/main.lua:
```diff
@@ -1486,6 +1553,44 @@ function Wallabag:setArchiveDirectory(touchmenu_instance)
     }:chooseDir()
 end
 
+--- A dialog used for setting a generic timeout value.
```
You can lose probably all of these comments. ;-)

Frenzie replied:
(I'll apply this, but it's quickest this way.)
```suggestion
```

Frenzie commented on plugins/wallabag.koplugin/main.lua:
```diff
@@ -1486,6 +1553,44 @@ function Wallabag:setArchiveDirectory(touchmenu_instance)
+                    end,
+                },
+                {
+                    text = _("Apply"),
```
```suggestion
                    text = _("Set timeout"),
```

zikasak replied:
Replaced with "Set timeout"

Frenzie commented:
Fine by me in principle, but pinging @NiLuJe for a second pair or eyes/brain.

NiLuJe commented:
Fine by me, if it actually helps.

I'd perhaps want more user-friendly descriptions, though.

While the `total` timeouts do what they say on the tin, the `block` ones are a bit murkier. They roughly match what's used as the socket timeout (i.e., how long we'll wait for a request to succeed).

NiLuJe commented:
So, I dunno, something like "connection timeout" vs. "timeout" (or something better) might be dumbing it down enough for it to mooostly make sense.

i.e., if the issue is a slow server and not you trying to download a ginormous amount of data, the `block` timeouts are the ones you want to tweak.

hius07 commented:
Usually there is some kind of space before units.
https://github.com/koreader/koreader/blob/14d28483621f86038623a78133671bc53b2f3c9f/frontend/apps/reader/modules/readerhighlight.lua#L751
https://github.com/koreader/koreader/blob/14d28483621f86038623a78133671bc53b2f3c9f/frontend/ui/widget/spinwidget.lua#L149

zikasak commented:
I've added a space before unit in the menu and changed the apply button text. 
About changing block timeout to something more friendly. I agree it is not really friendly to the user but unfortunately, I don't have any ideas...

Frenzie commented on plugins/wallabag.koplugin/main.lua:
```diff
@@ -353,6 +359,67 @@ function Wallabag:addToMainMenu(menu_items)
+                        sub_item_table = {
+                            {
+                                text_func = function()
+                                    return T(_("File download block timeout: %1\u{202F}s"), self.file_block_timeout)
```
A normal space will do. Doing it this way won't work with localization.

Wrt to the string, I "file block" merely refers to the fact that it's about files. So just call it something like `Article download timeout` vs `API request timeout`.

zikasak replied:
changed

Frenzie commented on plugins/wallabag.koplugin/main.lua:
```diff
@@ -353,6 +359,67 @@ function Wallabag:addToMainMenu(menu_items)
+                        sub_item_table = {
+                            {
+                                text_func = function()
+                                    return T(_("Article download block timeout: %1 s"), self.file_block_timeout)
```
@NiLuJe I think you wanted to change "block timeout" to "connection timeout"?

zikasak replied:
@Frenzie I don't think having two "connection timeout" setting makes sense. But as this timeout is for data waiting... Probably, something like "Article download wait timeout" will work?

Frenzie replied:
The other one's the download timeout, not the (wait for) connection timeout.

zikasak replied:
looks like I was confused... Renamed

Frenzie commented on plugins/wallabag.koplugin/main.lua:
```diff
@@ -1512,6 +1617,11 @@ function Wallabag:saveSettings()
         offline_queue                  = self.offline_queue,
         use_local_archive             = self.use_local_archive,
         archive_directory             = self.archive_directory,
+        -- NEW Timeout settings
```
The unnecessary comments are still here, besides that lgtm.

zikasak replied:
removed

Frenzie reviewed (approved):
@NiLuJe ?
